src/storage/src/hummock/compactor/mod.rs

```diff
@@ -255,18 +259,29 @@ impl Compactor {
         }
 
         let mut buffered = stream::iter(compaction_futures).buffer_unordered(parallelism);
-        while let Some(future_result) = buffered.next().await {
-            match future_result.unwrap() {
-                Ok((split_index, ssts)) => {
-                    output_ssts.push((split_index, ssts));
+        loop {
+            tokio::select! {
+                _ = &mut shutdown_rx => {
```

Comment:
Could you please explain how many benefits external cancellation will bring, with its implementation complexity introduced?

Reply:
There's only one conceivable benefit, which is early stopping of tasks on compactor side if meta side deems task dead.

Reply:
I believe the change is small compared to the rest of the PR (which is reliable heartbeats).

Reply:
In hindsight, it could have been seperated into a separate PR for simplicity's sake.